document/core/valid/instructions.rst

```diff
@@ -232,6 +232,439 @@ Reference Instructions
      C \vdashinstr \REFASNONNULL : [(\REF~\NULL~\X{ht})] \to [(\REF~\X{ht})]
    }
 
+.. _valid-ref.eq:
+
+:math:`\REFEQ`
+..............
+
+* The instruction is valid with type :math:`[(\REF~\NULL~\EQT) (\REF~\NULL~\EQT)] \to [\I32]`.
+
+.. math::
+   \frac{
+   }{
+     C \vdashinstr \REFEQ : [(\REF~\NULL~\EQT)~(\REF~\NULL~\EQT)] \to [\I32]
+   }
+
+.. _valid-ref.test:
+
+:math:`\REFTEST~\X{rt}`
+.......................
+
+* The :ref:`reference type <syntax-reftype>` :math:`\X{rt}` must be :ref:`valid <valid-reftype>`.
+
+* Then the instruction is valid with type :math:`[\X{rt}'] \to [\I32]` for any :ref:`valid <valid-reftype>` :ref:`reference type <syntax-reftype>` that :ref:`matches <match-reftype>` :math:`\X{rt}`.
```

Comment:
Is it fine to leave it implicit that the mentioned "valid reference type that matches `rt`" is `rt'`?